client/util.go

```diff
@@ -0,0 +1,116 @@
+package client
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/binary"
+	"errors"
+	"io"
+
+	sserver1 "github.com/keybase/search/protocol/sserver"
+	"golang.org/x/crypto/nacl/secretbox"
+)
+
+// The length of the overhead added to padding.
+const padPrefixLength = 4
+
+// pathnameToDocID encrypts a `pathname` to a document ID using `key`.
+func pathnameToDocID(pathname string, key *[32]byte) (sserver1.DocumentID, error) {
+	var nonce [24]byte
+	_, err := rand.Read(nonce[:])
+	if err != nil {
+		return sserver1.DocumentID(""), err
+	}
+
+	paddedPathname, err := padPathname(pathname)
+	if err != nil {
+		return sserver1.DocumentID(""), err
+	}
+
+	sealedBox := secretbox.Seal(nil, (paddedPathname), &nonce, key)
+
+	docIDRaw := append(nonce[:], sealedBox...)
```

Comment:
Might be good to put a 2-byte version in front of the nonce, and start it out at `00`.  That way if we ever need to upgrade the format of the doc ID and we already have a bunch of data we can't lose, we'll be able to tell different versions apart.